Provides hint to the video decoder that it would be OK to use a low-power mode that can not decode faster than 1x realtime.

jyavenard opened the pull request:
#### 54dd824eccb93445742e5ca875b2ea7ee77987ef
<pre>
Provides hint to the video decoder that it would be OK to use a low-power mode that can not decode faster than 1x realtime.
<a href="https://bugs.webkit.org/show_bug.cgi?id=294245">https://bugs.webkit.org/show_bug.cgi?id=294245</a>
<a href="https://rdar.apple.com/152930386">rdar://152930386</a>

Reviewed by Youenn Fablet.

If playback rate is between 0.9 and 1.1, set kVTDecodeFrame_1xRealTimePlayback
when decoding the CMSampleBuffer.
We now pass a OptionSet&lt;DecodingFlag&gt; to the WebCoreDecompressionSession to allow
more future expandability.

Add macro to only log if the level is set to debug. This avoids unecessarily
building the log string when the result won&apos;t end up being used.

No change in JS observable behaviour.

* Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.mm:
(WebCore::VideoMediaSampleRenderer::maybeBecomeReadyForMoreMediaData): Rename methods to be consistent with other naming style
(WebCore::VideoMediaSampleRenderer::stopRequestingMediaData): Rename methods to be consistent with other naming style
(WebCore::VideoMediaSampleRenderer::decodeNextSampleIfNeeded):
(WebCore::VideoMediaSampleRenderer::initializeDecompressionSession):
(WebCore::VideoMediaSampleRenderer::requestMediaDataWhenReady):
(WebCore::VideoMediaSampleRenderer::resetReadyForMoreMediaData): Rename method to be consistent with other naming style
(WebCore::VideoMediaSampleRenderer::resetReadyForMoreSample): Deleted.

Canonical link: <a href="https://commits.webkit.org/296081@main">https://commits.webkit.org/296081@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c05332d9fdb9ccabf241993f1bb589bb79c446e1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107324 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27006 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/17410 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/112538 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/57859 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/109288 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/27687 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/35506 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81471 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110253 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/21934 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96729 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61845 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21372 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/14859 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57303 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91302 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14893 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/115639 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/34390 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/25343 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90513 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/34766 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92979 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90249 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23006 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35157 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12938 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/30122 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34312 "Built successfully") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/39845 "Failed to build and analyze WebKit") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34058 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/37413 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/35719 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.mm:
```diff
@@ -285,22 +293,26 @@ static OSStatus sampleCallback(CMBufferRef buffer, void* refcon)
     assertIsMainThread();
 
-    m_readyForMoreSampleFunction = nil;
+    m_readyForMoreMediaDataFunction = nil;
```
```suggestion
    m_readyForMoreMediaDataFunction = nullptr;
```
?

jyavenard committed:
Provides hint to the video decoder that it would be OK to use a low-power mode that can not decode faster than 1x realtime.
https://bugs.webkit.org/show_bug.cgi?id=294245
rdar://152930386

Reviewed by Youenn Fablet.

If playback rate is between 0.9 and 1.1, set kVTDecodeFrame_1xRealTimePlayback
when decoding the CMSampleBuffer.
We now pass a OptionSet<DecodingFlag> to the WebCoreDecompressionSession to allow
more future expandability.

Add macro to only log if the level is set to debug. This avoids unecessarily
building the log string when the result won't end up being used.

No change in JS observable behaviour.

* Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.mm:
(WebCore::VideoMediaSampleRenderer::maybeBecomeReadyForMoreMediaData): Rename methods to be consistent with other naming style
(WebCore::VideoMediaSampleRenderer::stopRequestingMediaData): Rename methods to be consistent with other naming style
(WebCore::VideoMediaSampleRenderer::decodeNextSampleIfNeeded):
(WebCore::VideoMediaSampleRenderer::initializeDecompressionSession):
(WebCore::VideoMediaSampleRenderer::requestMediaDataWhenReady):
(WebCore::VideoMediaSampleRenderer::resetReadyForMoreMediaData): Rename method to be consistent with other naming style
(WebCore::VideoMediaSampleRenderer::resetReadyForMoreSample): Deleted.

Canonical link: https://commits.webkit.org/296081@main
